base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class Basedecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
```

Comment:
- Just as the previous reviewer suggested, please define the :nameable attribute as both a getter and setter i.e you can use the attr_accessor :nameable for that. This will make that variable to be easily accessible (read/write) by outside users without having to write class accessor methods